x/stakeibc/types/message_restore_interchain_account.go

```diff
@@ -0,0 +1,51 @@
+package types
+
+import (
+	"github.com/Stride-Labs/stride/utils"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+const TypeMsgRestoreInterchainAccount = "register_interchain_account"
+
+var _ sdk.Msg = &MsgRestoreInterchainAccount{}
+
+func NewMsgRestoreInterchainAccount(creator string, chainId string, accountType ICAAccountType) *MsgRestoreInterchainAccount {
+	return &MsgRestoreInterchainAccount{
+		Creator:     creator,
+		ChainId:     chainId,
+		AccountType: accountType,
+	}
+}
+
+func (msg *MsgRestoreInterchainAccount) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgRestoreInterchainAccount) Type() string {
+	return TypeMsgRestoreInterchainAccount
+}
+
+func (msg *MsgRestoreInterchainAccount) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
+}
+
+func (msg *MsgRestoreInterchainAccount) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgRestoreInterchainAccount) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
+	}
+	if err := utils.ValidateAdminAddress(msg.Creator); err != nil {
```

Comment:
I think we should allow anyone to restore this - there's no reason to restrict it (afaict) and I think we should remove ADMINS altogether before launch

Reply:
@asalzmann can you auth with axolo so that these messages show up from your account , just slightly more clear [](edited-with-Axolo)

Reply:
just pushed up a commit with the PR changes address, let me know if I have your blessing [](edited-with-Axolo)

Reply:
running the integration tests now too [](edited-with-Axolo)

Reply:
I've found the axolo channels useful so far for chatting about specific PRs internally with other reviewers , and only putting code comments on the PR itself on GH [](edited-with-Axolo)

Reply:
I'm down to try axolo! Have just noticed that tools outside of GH tend to get used less over time (maybe slack is integrated enough though that we'll use this) [](edited-with-Axolo)

Reply:
yeah down to test drive it and ditch it if it gets to use [](edited-with-Axolo)

Reply:
agreed. fwiw, I've liked it a lot so far. I'm much more up to date on the status of these PRs then I used to be

Reply:
I am too, nice to have the ping on slack rather than having to swap between slack and email [](edited-with-Axolo)

Reply:
the other minor comment I'll make is that I think we should try to push convos to GH for posterity (ik we can unarchive slack, but it's less ideal - if we lose access to slack or if someone else wants to understand why we made certain decisions, it's less transparent) [](edited-with-Axolo)